src/test/java/org/openrefine/extensions/commons/importer/RelatedCategoryFetcherTest.java

```diff
@@ -0,0 +1,90 @@
+package org.openrefine.extensions.commons.importer;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import okhttp3.HttpUrl;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+public class RelatedCategoryFetcherTest {
+
+    @Test
+    public void testGetRelatedCategories() throws Exception {
+
+        try (MockWebServer server = new MockWebServer()) {
+            server.start();
+            HttpUrl url = server.url("/w/api.php");
+            String jsonResponse = "{\"batchcomplete\":\"\",\"query\":{\"pages\":{\"127722\":"
+                    + "{\"pageid\":127722,\"ns\":6,\"title\":\"File:LasTres.jpg\","
+                    + "\"categories\":[{\"ns\":14,\"title\":\"Category:Costa Rica\"},"
+                    + "{\"ns\":14,\"title\":\"Category:Cute dogs\"},{\"ns\":14,\"title\":\"Category:Costa Rican dogs\"}]},"
+                    + "\"127752\":{\"pageid\":127752,\"ns\":6,\"title\":\"File:Pejiballes.jpg\","
+                    + "\"categories\":[{\"ns\":14,\"title\":\"Category:Costa Rica\"},"
+                    + "{\"ns\":14,\"title\":\"Category:Yummy food\"},{\"ns\":14,\"title\":\"Category:Costa Rican dishes\"}]}}}}";
+            server.enqueue(new MockResponse().setBody(jsonResponse));
```

Comment:
Here, you are preparing one mocked response for the first request made to the `MockWebServer`. If you anticipate that your code will be doing more requests, you should enqueue more responses accordingly (and similarly for the other tests).